### PR TITLE
[specific ci=1-27-Docker-Login] Fail quickly on docker login against harbor registry with incorrect credentials

### LINKED ISSUE #5195: Docker login responds with poor message with bad user/pass
As a VIC user, I should be able to use a bad password with docker login and get a message that the username or password was incorrect.

---

Standard docker returns the following message when the user cannot be authenticated,

```
Error response from daemon: Get https://registry-1.docker.io/v2/: unauthorized: incorrect username or password
```

This is what VIC responds:
```
Error response from daemon: download cancelled during retry delay
```

There's a bug in our OAuth code for bad user/pass.

### PR DESCRIPTION
**Pull request by willsherwood**
Fixes #5195

### CODE REVIEW

File: tests/test-cases/Group1-Docker-Commands/1-27-Docker-Login.robot [hunk 1]
```diff
@@ -34,5 +41,20 @@ Docker login and pull from docker.io
...
+    Set Environment Variable  HARBOR_IP  %{HARBOR-IP}
+    
+    Install Harbor Self Signed Cert
+    Install VIC Appliance To Test Server  vol=default --registry-ca=/etc/docker/certs.d/%{HARBOR-IP}/ca.crt  certs=${false}
+    
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} login --username=%{TEST_USERNAME} --password=%{TEST_PASSWORD}
+    Should Contain  ${output}  Login Succeeded
+    Should Be Equal As Integers  ${rc}  0
+
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} login --username=%{TEST_USERNAME} --password=incorrectPassword
```
Comment (mhagen-vmware) [inline tests/test-cases/Group1-Docker-Commands/1-27-Docker-Login.robot]: incorrectPassword -> %{TEST_USERNAME}1

File: tests/test-cases/Group1-Docker-Commands/1-27-Docker-Login.robot [hunk 2]
```diff
@@ -34,5 +41,20 @@ Docker login and pull from docker.io
...
+    Should Contain  ${output}  incorrect username or password
+    Should Be Equal As Integers  ${rc}  1
+
```
Comment (mhagen-vmware) [inline tests/test-cases/Group1-Docker-Commands/1-27-Docker-Login.robot]: A test for truncated password (in addition to the plus 1 test above) and equivalent username tests please.

**Comment by sflxn** [comment]
Will, you also need to update the Login.md file.

File: tests/test-cases/Group1-Docker-Commands/1-27-Docker-Login.robot (hunk 1 above)
Comment (willsherwood) [inline tests/test-cases/Group1-Docker-Commands/1-27-Docker-Login.robot]: Do you mean %{TEST_PASSWORD}1?

**Comment by willsherwood** [comment]
I will write out the .md as soon as the test is functioning as desired.

**Comment by willsherwood** [comment]
Closing this prior to leaving: pending an integration test. The Harbor robot utility package currently does not function properly to copy server certificates since the removal of Harbor UI. When Admiral provides the necessary API for accessing Harbor instances, the test should be trivial to write:

1) Start an instance of harbor, or use an existing one
2) Login with an incorrect password and verify that an 'invalid username or password' message is shown, and not another obscure error.
3) Login with a correct password and verify that we can correctly pull a private image.

**Comment by mdubya66** [comment]
we can add an integration test later for this. approved for merge once the build has succeded.

**Pull request merged**